test/unit/JsonService.test.ts

```diff
@@ -30,30 +29,41 @@ describe("JsonService", () => {
             try { await p; } catch {}
         });
 
-        it("should make GET request to url", async () => {
-            // act
-            const p = subject.getJson("http://test");
+        it("should reject promise when no url is passed", async () => {
+            try {
+                // act
+                await subject.getJson("");
+            } catch (err) {
+                // assert
+                expect(err).toBeInstanceOf(Error);
+                expect((err as Error).message).toContain("url");
+            }
+        });
 
-            // assert
-            expect(fetchMock).toBeCalledWith("http://test", {
-                headers: {},
-                method: "GET"
-            });
-            // eslint-disable-next-line no-empty
-            try { await p; } catch {}
+        it("should make GET request to url", async () => {
+            try {
+                // act
+                await subject.getJson("http://test");
+            } catch {
+                // assert
+                expect(fetchMock).toBeCalledWith("http://test", {
+                    headers: {},
+                    method: "GET"
+                });
+            }
         });
 
         it("should set token as authorization header", async () => {
-            // act
-            const p = subject.getJson("http://test", "token");
-
-            // assert
-            expect(fetchMock).toBeCalledWith("http://test", {
-                headers: { Authorization: "Bearer token" },
-                method: "GET"
-            });
-            // eslint-disable-next-line no-empty
-            try { await p; } catch {}
+            try {
+                // act
+                await subject.getJson("http://test", "token");
+            } catch {
+                // assert
+                expect(fetchMock).toBeCalledWith("http://test", {
+                    headers: { Authorization: "Bearer token" },
+                    method: "GET"
+                });
```

Comment:
This test will create a false positive if `.getJson` unexpectedly succeeds. Use something like this for testing async errors:

```suggestion
                await expect(subject.getJson("http://test", "token")).rejects.toThrow(Error);
                expect(fetchMock).toBeCalledWith("http://test", {
                    headers: { Authorization: "Bearer token" },
                    method: "GET"
                });